src/attributes.rs

```diff
@@ -94,12 +94,18 @@ impl<'a> AttributeMetaItem<'a> {
 
             if !inside_string_literal {
                 if Self::is_left_bracket(c) {
-                    bracket_depth += 1;
+                    brackets.push(c);
                 } else if Self::is_right_bracket(c) {
-                    bracket_depth -= 1;
+                    if let Some(top_left) = brackets.pop() {
+                        if Self::matching_right_bracket(top_left) != c {
+                            return None;
+                        }
+                    } else {
+                        return None;
+                    }
```

Comment:
A comment explaining the semantic meaning of `return None` would help explain what's going on here.
```suggestion
                    // If the brackets don't match in any way, give up on parsing 
                    // individual arguments since we don't understand the format.
                    if let Some(top_left) = brackets.pop() {
                        if Self::matching_right_bracket(top_left) != c {
                            return None;
                        }
                    } else {
                        return None;
                    }
```